lte/gateway/c/session_manager/SessionCredit.cpp

```diff
@@ -150,18 +168,24 @@ bool SessionCredit::is_quota_exhausted(float threshold) const {
   if (credit_limit_type_ != FINITE) {
     return false;
   }
-  uint64_t total_reported = buckets_[REPORTED_TX] + buckets_[REPORTED_RX];
-  uint64_t total_usage    = buckets_[USED_TX] + buckets_[USED_RX];
 
   bool rx_exhausted = compute_quota_exhausted(
-      buckets_[ALLOWED_RX], buckets_[REPORTED_RX], buckets_[USED_RX],
-      threshold);
+      buckets_[ALLOWED_RX],  buckets_[USED_RX],
+      threshold,
+      buckets_[ALLOWED_RX_FLOOR]);
   bool tx_exhausted = compute_quota_exhausted(
-      buckets_[ALLOWED_TX], buckets_[REPORTED_TX], buckets_[USED_TX],
-      threshold);
+      buckets_[ALLOWED_TX],  buckets_[USED_TX],
+      threshold,
+      buckets_[ALLOWED_TX_FLOOR]);
+  bool total_exhausted = compute_quota_exhausted(
+      buckets_[ALLOWED_TOTAL], buckets_[USED_TX] + buckets_[USED_RX],
+      threshold, buckets_[ALLOWED_TOTAL_FLOOR]);
 
   bool is_exhausted = false;
   switch (grant_tracking_type_) {
+    case ALL_TOTAL_TX_RX:
```

Comment:
Ok so we currently cover 4 cases right?
1. All three fields
2. only total 
3. both rx and tx
4. only rx
5. only tx

This should cover both gx and gy needs. I'm guessing your opinion is that these are the ones we need for now? (I think so)

Reply:
I think thats right. Per what i can see many will just use `TOTAL_ONLY`. But the protocol allows `ALL_TOTAL_TX_RX` too. So lets use for now those 4 options.